source/common/common/dump_state_utils.h

```diff
@@ -9,11 +9,26 @@ namespace Envoy {
 // i.e. under the Envoy signal handler if encountering a crash due to OOM, where allocating more
 // memory would likely lead to the crash handler itself causing a subsequent OOM.
 
-#define DUMP_MEMBER(member) ", " #member ": " << (member)
+#define _DUMP_MEMBER(member) ", " #member ": " << (member)
+#define _DUMP_MEMBER_VIA_VALUE(member, value) ", " #member ": " << (value)
+#define _DUMP_MEMBER_SELECTOR(_1, _2, DUMP_MACRO, ...) DUMP_MACRO
+
+// This is a workaround for fact that MSVC expands __VA_ARGS__ after passing them into a macro,
+// rather than before passing them into a macro. Without this,
+// _DUMP_MEMBER_SELECTOR does not work correctly when compiled with MSVC.
+#define EXPAND(X) X
+
+// If DUMP_MEMBER is called with one argument, then _DUMP_MEMBER is called.
+// If DUMP_MEMBER is called with two arguments, then _DUMP_MEMBER_VIA_VALUE is called.
+#define DUMP_MEMBER(...)                                                                           \
+  EXPAND(_DUMP_MEMBER_SELECTOR(__VA_ARGS__, _DUMP_MEMBER_VIA_VALUE, _DUMP_MEMBER)(__VA_ARGS__))
```

Comment:
I'm not seeing any uses of the 2 argument DUMP_MEMBER macro.  I think this was done prior to the introduction of DUMP_NULLABLE_MEMBER below